functions/src/actions/wayback-machine.js

```diff
@@ -0,0 +1,122 @@
+// Third party imports
+const axios = require('axios');
+const mustache = require('mustache');
+const xml2js = require('xml2js');
+
+// Local imports
+const config = require('../config');
+const {debug} = require('../utils/logger')('ia:actions:wayback-machine');
+const dialog = require('../dialog');
+const endpointProcessor = require('../network/endpoint-processor');
+const traverse = require('../utils/traverse');
+const waybackStrings = require('../strings').intents.wayback;
+
+/**
+ * Handle wayback query action
+ * - fill slots of wayback query
+ * - perform data requests to archive and alexa rankings
+ * - construct response speech for action
+ *
+ * @param app
+ */
+function handler (app) {
+  // Create wayback object
+  const waybackObject = {
+    url: '',
+    earliestYear: 0,
+    latestYear: 0,
+    totalUniqueURLs: 0,
+    alexaWorldRank: 0,
+    alexaUSRank: 0,
+    speech: waybackStrings.default,
+  };
+
+  // Check to see that both parameters have content
+  if (!app.params.getByName('wayback') && !app.params.getByName('url')) {
+    debug('wayback action called by mistake');
+    dialog.ask(app, waybackObject);
+  }
+
+  // Get url parameter and make url queries
+  waybackObject.url = app.params.getByName('url');
+  const archiveQueryURL = endpointProcessor.preprocess(
+    config.wayback.ARCHIVE, app, waybackObject
+  );
+  const alexaQueryURL = endpointProcessor.preprocess(
+    config.wayback.ALEXA, app, waybackObject
+  );
+
+  return Promise.all([axios.get(archiveQueryURL), axios.get(alexaQueryURL)])
+    .then(function (allData) {
+      // All data available here in the order it was called.
+
+      // Parse data from archive request
+      let archiveJSON = allData[0].data;
+      archiveEngine(archiveJSON, waybackObject);
+
+      // Parse data from alexa request
+      let alexaJSON;
+      let XMLparser = new xml2js.Parser();
+      let convertXML = new Promise((resolve, reject) => {
+        XMLparser.parseString(allData[1].data, function (err, result) {
+          if (err) {
+            debug('The XML parser didn\'t work');
+            waybackObject.speech = waybackStrings.error;
+            dialog.ask(app, waybackObject);
+          } else {
+            alexaJSON = JSON.parse(JSON.stringify(result));
```

Comment:
it seems that you forgot to run `resolve` here, and btw, you don't need `alexaJSON` here, because you can just pass a result to `resolve`.